scripts/fires/wfigs_data.py

```diff
@@ -0,0 +1,278 @@
+    def get_cost(cost_val):
+        if cost_val and not np.isnan(cost_val):
+            return _EXPECTED_LOSS_MAP.format_map({'costs': cost_val})
+        else:
+            return None
+
+    # Write fire area in an ingest-able format.
+    def get_area(area, label, mapping):
+        if area and not np.isnan(area):
+            return mapping.format_map({label: area})
+        else:
+            return None
+
+    # Get cause-dcid for the fire cause.
+    def get_cause(cause_str):
+        if cause_str:
+            cause = cause_str.replace(" ", "").replace("/", "Or")
+            if cause in mapping.CAUSE_DCID_MAP:
+                cause = mapping.CAUSE_DCID_MAP[cause]
+            return cause
+        else:
+            return None
+
+    # get personnel count for the incident.
+    def get_personnel(personnel_val):
+        if personnel_val and not np.isnan(personnel_val):
+            return personnel_val
+        else:
+            return None
+
+    # If a parent fire exists, associate it with given fire.
+    def get_parent_fire(row):
+        if not (np.isnan(row.IsCpxChild)) and row.IsCpxChild == 1 and row.CpxID:
+            # remove curly brackets from front and end of string.
+            return '{prefix}{id}'.format(prefix=_FIRE_DCID_PREFIX,
+                                         id=row.CpxID[1:-1].lower())
+        else:
+            return None
+
+    df["FireDiscoveryDateTime"] = df["FireDiscoveryDateTime"].apply(
+        get_datetime)
+    df["Year"] = df["FireDiscoveryDateTime"].apply(lambda x: x.year)
+    df["FireDiscoveryDateTime"] = df["FireDiscoveryDateTime"].apply(
+        get_date_str)
+    df = df[df["Year"] >= 2014]
+    df["ContainmentDateTime"] = df["ContainmentDateTime"].apply(
+        get_datetime).apply(get_date_str)
+    df["ControlDateTime"] = df["ControlDateTime"].apply(get_datetime).apply(
+        get_date_str)
+    df["InitialResponseDateTime"] = df["InitialResponseDateTime"].apply(
+        get_datetime).apply(get_date_str)
+    df["Costs"] = df["EstimatedCostToDate"].apply(get_cost)
+    df['BurnedArea'] = df['DailyAcres'].apply(get_area,
+                                              args=('daily_acres',
+                                                    _DAILY_ACRES_MAP))
+    df["FireCause"] = df["FireCause"].apply(get_cause)
+    df["FireCauseGeneral"] = df["FireCauseGeneral"].apply(get_cause)
+    df["FireCauseSpecific"] = df["FireCauseSpecific"].apply(get_cause)
+    df["TotalIncidentPersonnel"] = df["TotalIncidentPersonnel"].apply(
+        get_personnel)
+    df["ParentFire"] = df.apply(get_parent_fire, axis=1)
+    df['InitialResponseAcres'] = df['InitialResponseAcres'].apply(
+        get_area, args=('initial_response_area', _INITIAL_RESPONSE_AREA_MAP))
+
+    df["Location"] = df.apply(get_place, axis=1)
+    df["FIPS"] = df.apply(get_fips, axis=1)
+    df["Location"] = df["Location"].fillna(df["FIPS"])
+    df.loc[df["Location"] == '', 'Location'] = df["FIPS"]
```

Comment:
would be cleaner if fillna above filled in with '', then values are replaced only once to FIPS.

also, the length of `df.loc[...]` would not match the length of`df['FIPS']`... are the right cells being used for the fill?

Reply:
Yes, this would only replace values where the condition is met with corresponding FIPS value

Reply:
Done